context-propagation/src/main/java/io/smallrye/opentracing/SmallRyeOpenTracingContextProvider.java

```diff
@@ -0,0 +1,43 @@
+package io.smallrye.opentracing;
+
+import java.util.Map;
+
+import org.eclipse.microprofile.context.spi.ThreadContextProvider;
+import org.eclipse.microprofile.context.spi.ThreadContextSnapshot;
+
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+
+public class SmallRyeOpenTracingContextProvider implements ThreadContextProvider {
+
+    @Override
+    public ThreadContextSnapshot currentContext(Map<String, String> props) {
+        Tracer activeTracer = GlobalTracer.get();
```

Comment:
nit: move this to a final static class field.

Reply:
currentContext is called, when submitting a task to a ManagedExecutorService, so currentContext my be called in parallel by multiple threads. I don't think putting it an (volatile) static field is the best solution